StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeRangeLifeSteal.cpp

```diff
@@ -0,0 +1,45 @@
+#include "UpgradeRangeLifeSteal.h"
+#include "EngineMath.h"
+#include "ECS/Entity.h"
+#include "Player.h"
+#include "Components/ProjectileComponent.h"
+
+UpgradeRangeLifeSteal::UpgradeRangeLifeSteal(Entity* parent)
+	:Upgrade(parent)
+{
+	// name the upgrade!
+	SetName("UpgradeRangeLifeSteal");
+	// set the upgrade type/types!
+	SetType(F_UpgradeType::RANGE);	// goes on the projectiles
+	// set the price of this upgrade
+	m_Price = 1;
+	m_StartingPrice = m_Price;
+	// set short description 
+	m_Description = "Converts a percentage of range damage done into own health. 10\% for the first level, 5\% more per upgrade level";
+	// percentage of damage done to steal as life
+	m_PercentageGain = 0.10;
+}
+
+UpgradeRangeLifeSteal::~UpgradeRangeLifeSteal()
+{
+}
+
+void UpgradeRangeLifeSteal::IncreaseLevel()
+{
+	m_Level++;
+	m_PercentageGain += 0.05;
+	m_Price *= 2;
+}
+
+void UpgradeRangeLifeSteal::OnRangedHit()
+{
+	// get damage from projectile
+	int damage = m_pParentEntity->GetComponent<component::ProjectileComponent>()->GetDamage();
+	// calculate amount of health to add
+	int hpChange = static_cast<int>(damage * m_PercentageGain);
+	// add a bit of health
+	Player::GetInstance().GetPlayer()->GetComponent<component::HealthComponent>()->ChangeHealth(hpChange);
+
+	// increase price for next level
+	m_Price *= 2;
```

Comment:
Increasing price of next level for every hit? Sounds expensive!

Reply:
oops :> fixed it!